packages/in_app_purchase/in_app_purchase_android/migration_guide.md

```diff
@@ -0,0 +1,143 @@
+# Migration Guide from 0.2.x to 0.3.0
+
+Starting November 2023, Android Billing Client V4 is no longer supported,
+requiring app developers using the plugin to migrate to 0.3.0.
+Version 0.3.0 introduces breaking changes to the API as a result of changes in
+the underlying Google Play Billing Library.
+For context around the changes see the [recent changes to subscriptions in Play Console][1]
+and the [Google Play Billing Library migration guide][2].
+
+## SKUs become Products
+
+SKUs have been replaced with products. For example, the `SkuDetailsWrapper`
+class has been replaced by `ProductDetailsWrapper`, and the `SkuType` enum has
+been replaced by `ProductType`.
+
+Products are used to represent both in-app purchases (also referred to as
+one-time purchases) and subscriptions. There are a few changes to the data model
+that are important to note.
+
+Previously, an SKU of `SkuType.subs` would contain a `freeTrialPeriod`,
+`introductoryPrice` and some related properties. In the new model, this has been
+replaced by a more generic approach.
+A subscription product can have multiple pricing phases, each with its own
+price, subscription period etc. This allows for more flexibility in the pricing
+model.
+
+In the next section we will look at how to migrate code that uses the old model
+to the new model. Instead of listing all possible use cases, we will focus on
+three: getting the price of a one time purchase, and handling both free trials
+and introductory prices for subscriptions. Other use cases can be migrated in a
+similar way.
+
+### Use case: getting the price of a one time purchase
+
+Below code shows how to get the price of a one time purchase before and after
+the migration. Other properties can be obtained in a similar way.
+
+Code before migration:
+
+```dart
+SkuDetailsWrapper sku;
+
+if (sku.type == SkuType.inapp) {
+  String price = sku.price;
+}
+```
+
+Code after migration:
+
+```dart
```

Comment:
Although it's not enforced by CI tooling for files other than README.md, we should use [code excerpts](https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#readme-code) for all of the "after" examples so that we are certain that they compile, and continue to compile as the package changes. I believe once you do the setup described there it will work for any `.md` file, not just README.

(If that's not the case please let me know, and I'll update the tooling as necessary.)

Reply:
Thanks for pointing me to the existence of code excerpts, I like it. I was able to generate excerpts for README files, but not for the migration guide `.md` file. Could you check for me if the tooling allows for any `.md` file?

Below is (part of) the output of `dart run script/tool/bin/flutter_plugin_tools.dart update-excerpts --packages in_app_purchase`:

```
Running command: "dart run code_excerpt_updater --write-in-place --yaml --no-escape-ng-interpolation ../README.md" in /Users/jeroen/Projects/flutter-packages/packages/in_app_purchase/in_app_purchase_android/example
Processed 1 Dart/Jade/Markdown files: 0 out of 1 fragments needed updating.
Running command: "dart run code_excerpt_updater --write-in-place --yaml --no-escape-ng-interpolation README.md" in /Users/jeroen/Projects/flutter-packages/packages/in_app_purchase/in_app_purchase_android/example
Processed 1 Dart/Jade/Markdown files: 0 out of 0 fragments needed updating.
```

It looks like it is hardcoded to cater to `README.md` and `../README.md`.

Reply:
Oops, so it is. I tried to prescreen in the tool source, and looked for the wrong thing. I'll take a look now.

Reply:
https://github.com/flutter/packages/pull/4212 will fix it; once that lands you can merge/rebase. Sorry about that!

Reply:
Now it should work :)